### ISSUE TITLE
Bounty ships keep landing on governments who offered the bounty

### ISSUE THREAD

**Issue by PeacefulPotato** [endless-sky/endless-sky#6527]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When I select the "Stolen ship near ..." missions, the ships tend to enter the system and then immediately land on the planet that offered the mission. This makes absolutely no sense as that would just lead to them being caught by the authorities, and makes the missions near impossible to complete since the ships never leave again, whereas if they flee to another system at least you can follow them (and eventually they will run out of fuel).

### Steps to Reproduce

1. Accept a "stolen ship near..." mission
2. Watch the ship arrive and then land on the planet you just left

### Expected Behavior

The ship should always attempt to jump out of the system if possible, and should never be able to land unless they are in a government that didn't offer the bounty and does not have positive relations with said government. Doing otherwise would just be stupid.

### Link to save file

https://bin.snopyta.org/?97253a3358c08142#71F6yQBZm7cShJ8DNqkgPYGuYAM4q5QwYDnmrRsjScvd

### Operating System

Devuan Chimaera 4.0 + MX Linux 21

### Game Source

ESLauncher2

### Game Version

8add4d3a29a999e10a1d8e4b9dffa74fdd7bbc1e

**Comment by PeacefulPotato** [comment]
Also, one time an arrow without a ramscoop or fuel pod seemed to jump more than 4 times in quick succession, but that could just have been them landing.

**Comment by tibetiroka** [comment]
Can confirm on [ee1d8b6](https://github.com/endless-sky/endless-sky/commit/ee1d8b686cb22c858668d36df8ed90861eb99204). The bounty ship lands on a planet if:
 - the player's flagship is cloaked
 - it runs out of fuel
 - if the player is far away from the planet, even if not cloaked

In some cases it took me quite a while to trigger these specific behaviours.

Currently, it seems like that landing on the same planet and taking off doesn't force the bounty ship to take off with you _unless you reload the save_. Taking off multiple times without reloading the save does not help. That seems like an independent issue with the personalities, the missions themselves are not broken.

Since the provided save file is no longer available I'm uploading two versions here: 
 - Save game after the bounty and the player have landed: [Test Pilot.txt](https://github.com/endless-sky/endless-sky/files/9358212/Test.Pilot.txt)
 - Save game after accepting a bounty: [Test Pilot.txt](https://github.com/endless-sky/endless-sky/files/9358308/Test.Pilot.txt)

**Comment by EjoThims** [comment]
I don’t think it landing on same government is really a bug (even if it may seem counterintuitive), but it never taking off again without reloading is definitely a problem.
